corehq/apps/app_manager/tests/data/suite/child-module-entry-datums-added-usercase.xml

```diff
@@ -0,0 +1,30 @@
+<partial>
+  <entry>
+    <command id="m0-f0">
+      <text>
+        <locale id="forms.m0f0"/>
+      </text>
+    </command>
+    <instance id="casedb" src="jr://instance/casedb"/>
+    <instance id="commcaresession" src="jr://instance/session"/>
+    <session>
+      <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='gold-fish'][@status='open']" value="./@case_id" detail-select="m0_case_short" detail-confirm="m0_case_long"/>
+      <datum id="case_id_new_guppy_0" function="uuid()" />
+      <datum id="usercase_id" function="instance('casedb')/casedb/case[@case_type='commcare-user'][hq_user_id=instance('commcaresession')/session/context/userid][1]/@case_id"/>
+    </session>
+  </entry>
+  <entry>
+    <command id="m1-f0">
+      <text>
+        <locale id="forms.m1f0"/>
+      </text>
+    </command>
+    <instance id="casedb" src="jr://instance/casedb"/>
+    <instance id="commcaresession" src="jr://instance/session"/>
+    <session>
+      <datum id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='gold-fish'][@status='open']" value="./@case_id" detail-select="m0_case_short"/>
+      <datum id="case_id_new_guppy_0" function="uuid()" />
+      <datum id="case_id_guppy" nodeset="instance('casedb')/casedb/case[@case_type='guppy'][@status='open'][index/parent=instance('commcaresession')/session/data/case_id]" value="./@case_id" detail-select="m1_case_short"  detail-confirm="m1_case_long"/>
```

Comment:
To answer [your question][1] from the previous PR, @snopoke, yes, I think you're right. There should be a `usercase_id` datum. I'll check why it's not being copied here. ... Although, we wouldn't need it unless the form preloads or updates the usercase, right? ... That said, I don't like that it's a special case, so I agree that it should be here.

  [1]: https://github.com/dimagi/commcare-hq/commit/1d1066e2d645d2d4e600fe4845b6481b70139ebb#commitcomment-12208105